socs/Lakeshore/Lakeshore336.py

```diff
@@ -0,0 +1,1205 @@
+    '2': 'zone',
+    '3': 'open loop',
+    '4': 'monitor out',
+    '5': 'Warm up'
+}
+output_modes_lock = {v.lower(): k for k, v in output_modes_key.items()}
+
+channel_key = {
+    '0': 'none',
+    '1': 'A',
+    '2': 'B',
+    '3': 'C',
+    '4': 'D'
+}
+channel_lock = {v: k for k, v in channel_key.items()}
+
+heater_display_key = {
+    '1': 'current',
+    '2': 'power'
+}
+heater_display_lock = {v: k for k, v in heater_display_key.items()}
+
+max_current_key = {
+    '0': 'User',
+    '1': .707,
+    '2': 1.,
+    '3': 1.141,
+    '4': 2
+}
+max_current_lock = {v: k for k, v in max_current_key.items()}
+
+heater_range_key = {
+    "0": "off",
+    "1": "low",
+    "2": "medium",
+    "3": "high"
+}
+heater_range_lock = {v: k for k, v in heater_range_key.items()}
+
+ramp_key = {
+    '0': 'off',
+    '1': 'on'
+}
+ramp_lock = {v: k for k, v in ramp_key.items()}
+
+# main class - Lakeshore 336 driver
+
+
+class LS336:
+    """
+    Implements a lakeshore 336 box to interface with client scripts.
+    Only contains locally relevant information; namely, port parameters.
+    The state of the device is not stored locally to avoid the potential for inconsistent information.
+    Device status can always be accessed (accurately) through a msg.    
+    """
+    # Constructor and instance variables
+
+    def __init__(self, port, timeout=1.0):
+
+        # LS336 defaults
+        self.baud = 57600
+        self.bytesize = serial.SEVENBITS
+        self.parity = serial.PARITY_ODD
+        self.stopbits = serial.STOPBITS_ONE
+        self.term = '\r\n'
+
+        self.port = port
+        self.timeout = timeout
+        self.comm = serial.Serial(
+            self.port, self.baud, self.bytesize, self.parity, self.stopbits, timeout)
+        self.comm.close()
+
+        self.id = self.get_id()
+        print(self.id)  # print idenfitication information to see if working
+
+        # Get Channels
+        inps = ['A', 'B', 'C', 'D']
+        self.channels = {inp: Channel(self, inp) for inp in inps}
+
+        # Get Heaters
+        htrs = ['1', '2']
+        self.heaters = {out: Heater(self, out) for out in htrs}
+
+    # Instance methods
+
+    def msg(self, cmd, delay=0.1):
+        r"""
+        Sends character string cmd to the device port, including commands and queries.
+
+        Parameters
+        ------
+        cmd : string
+            The character sequence to send to the device. Must be in the device proprietary
+            library, availablein the lakeshore manual.
+
+        delay : float or int
+            After writing, the function will lock out the kernel for delay seconds.
+            Within a script, this prevents too many cmds being written to the device too quickly.
+
+        Returns
+        ------
+        out : str
+            If cmd is a query (ie, contains '?'), returns the device response. Otherwise returns ''.
+        """
+
+        assert delay > 0.05, 'msg delay must be greater than 50ms'
```

Comment:
I like the idea here, but if we encounter this it'll raise an AssertionError and stop the program. Maybe better would be to force delays shorter than 50 ms to be 50 ms and then log a warning about the provided delay. This ensures the msg still goes through with proper timing and the user is notified about their improper delay.

(We didn't make this a parameter in other Lakeshore drivers because this timing *has* to be respected, else communication isn't guaranteed to work properly.)